src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerHandler.cs

```diff
@@ -0,0 +1,95 @@
+    {
+        private ServerViewInfo serverViewInfo = new ServerViewInfo();
+        private ServerConfigService configService = new ServerConfigService();
+        private Server server = null;
+
+        public ServerHandler(ConnectionService connectionService) : base(connectionService)
+        {
+        }
+
+        public override bool CanHandleType(SqlObjectType objectType)
+        {
+            return objectType == SqlObjectType.Server;
+        }
+
+        public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRequestParams requestParams)
+        {
+            ConnectionInfo connInfo = this.GetConnectionInfo(requestParams.ConnectionUri);
+            ServerConnection serverConnection = ConnectionService.OpenServerConnection(connInfo, ObjectManagementService.ApplicationName);
+
+            using (var context = new ServerViewContext(requestParams, serverConnection))
+            {
+                var serverSmo = CreateServerSmoObject(context).Result;
+                if(serverSmo != null)
+                {
+                    this.serverViewInfo.ObjectInfo = new ServerInfo()
+                    {
+                        Name = serverSmo.Name,
+                        HardwareGeneration = serverSmo.HardwareGeneration,
+                        Language = serverSmo.Language,
+                        MemoryInMB = serverSmo.PhysicalMemory,
+                        OperatingSystem = serverSmo.HostDistribution,
+                        Platform = serverSmo.HostPlatform,
+                        Processors = serverSmo.Processors,
+                        IsClustered = serverSmo.IsClustered,
+                        IsHadrEnabled = serverSmo.IsHadrEnabled,
+                        IsPolyBaseInstalled = serverSmo.IsPolyBaseInstalled,
+                        IsXTPSupported = serverSmo.IsXTPSupported,
+                        Product = serverSmo.Product,
+                        ReservedStorageSizeMB = serverSmo.ReservedStorageSizeMB,
+                        RootDirectory = serverSmo.RootDirectory,
+                        ServerCollation = serverSmo.Collation,
+                        ServiceTier = serverSmo.ServiceTier,
+                        StorageSpaceUsageInGB = serverSmo.UsedStorageSizeMB,
+                        Version = serverSmo.Version.ToString(),
+                        MinServerMemory = configService.GetConfigByName(this.server, "min server memory (MB)").ConfigValue,
+                        MaxServerMemory = configService.GetConfigByName(this.server, "max server memory (MB)").ConfigValue
+                    };
+                }
+
+                return Task.FromResult(new InitializeViewResult { ViewInfo = this.serverViewInfo, Context = context });
+            }
+        }
+
+        public override Task Save(ServerViewContext context, ServerInfo serverInfo)
+        {
+            throw new NotSupportedException();
```

Comment:
Could you add some more descriptive errors for this exception and the one in Script? Something that would say this exception is coming from the server handler, so we don't get cryptic errors being reported to the frontend if someone happens to call either of these for a server.